domain/secretbackend/params.go

```diff
@@ -16,6 +16,14 @@ type BackendIdentifier struct {
 	Name string
 }
 
+// String returns the identifier as a string.
+func (i BackendIdentifier) String() string {
+	if i.ID != "" {
```

Comment:
Should we prefer name? If this is used in error messages etc, the name has more meaning to a user.